### PR TITLE
Add e2e pipeline

### PR DESCRIPTION
**Pull request by tdaines42**
## Why is this PR needed?

We want to run the testrunner pytest tests against prs. At this time though we're not sure if we want to disable the Ginkgo ones.

Fixes #

## What does this PR do?

Creates a new pipeline to run the testrunner pytest tests.

### CODE REVIEW

File: ci/Makefile [hunk 1]
```diff
@@ -80,3 +80,9 @@ test_upgrade_apply_from_previous:
 .PHONY: test_upgrade_apply_user_lock
 test_upgrade_apply_user_lock:
 	${TEST_RUNNER} -v vars.yaml -p ${PLATFORM} test --verbose --suite test_skuba_upgrade.py --test test_upgrade_apply_user_lock
+
+.PHONY: test_pr_e2e
+test_pr:
+	${TEST_RUNNER} -v vars.yaml -p ${PLATFORM} test --verbose --suite test_cilium.py
+	${TEST_RUNNER} -v vars.yaml -p ${PLATFORM} test --verbose --suite test_deployments.py
+	${TEST_RUNNER} -v vars.yaml -p ${PLATFORM} test --verbose --suite test_skuba_upgrade.py
```
Comment (MaximilianMeister) [inline ci/Makefile]: if this runs against every PR i would only run 1 upgrade test from the suite:

```
${TEST_RUNNER} -v vars.yaml -p ${PLATFORM} test --verbose --suite test_skuba_upgrade.py --test test_upgrade_plan_from_previous
```

Comment (kravciak) [reply ci/Makefile]: `test_cilium.py` is not 100% stable until https://github.com/SUSE/skuba/blob/master/ci/infra/testrunner/tests/test_cilium.py#L16 is replaced.

**Comment by pablochacin** [comment]
> this might take a bit of time and i'm not really sure if we should always run the tests for every PR

@MaximilianMeister I agree we shouldn't. This e2e pipeline should be run periodically against master, not with each PR. 

As we don't have at this point narrow scope integration tests [1] we should probably use one or  two e2e tests which offer a broad code coverage as part of the PR pipeline. We already have one in ginkgo. We can keep using it or replace with something similar in testrunner like the deployments tests. IMHO this is not important. 

[1]  https://martinfowler.com/bliki/IntegrationTest.html

**Comment by MaximilianMeister** [comment]
> I think this pipeline should run nightly. Not per PR.

nightly e2e tests are already running here https://ci.suse.de/view/CaaSP/job/caasp-jobs/job/e2e/

if there are missing ones, they need to be added to this folder

**Comment by MaximilianMeister** [comment]
> if there are missing ones, they need to be added to this folder

not all of them were enabled yet, i have done that now in https://github.com/SUSE/skuba/pull/624

**Pull request closed without merging**